lib/spack/spack/solver/concretize.lp

```diff
@@ -480,6 +484,23 @@ error(2, "Attempted to use external for '{0}' which does not satisfy any configu
 variant(Package, Variant) :- variant_condition(ID, Package, Variant),
                              condition_holds(ID).
 
+% propagate the variant
+variant_value(Descendant, Variant, Value) :-
+  node(Package), path(Package, Descendant),
+  variant(Package, Variant),
+  variant(Descendant, Variant),
+  variant_value(Package, Variant, Value),
+  variant_propagate(Package, Variant),
+  not variant_set(Descendant, Variant),
+  variant_possible_value(Descendant, Variant, Value).
+
+
+:- variant_propagate(Package, Variant),
+   variant_propagate(Descendant, Variant),
+   path(Package, Descendant),
+   build(Package),
+   error("Cannot have variant propagation on parent and descendant"). % Only if the same variant
```

Comment:
This constraint needs to be rewritten to infer an error, rather than failing the solve. I think the issue is that you did it the old way before the error handling PR changed things, and it needs to change to match other constraints.

I think the next constraint is a good model for this, something like 
```suggestion
error(2, "{0} and dependency {1} cannot both propagate variant '{2}'", Package, Descendant, Variant) :- 
  variant_propagate(Package, Variant),
  variant_propagate(Descendant, Variant),
  path(Package, Descendant),
  build(Package).
```